freud/box.pyx

```diff
@@ -301,9 +302,13 @@ cdef class Box:
         vecs = np.atleast_2d(vecs)
         vecs = freud.util._convert_array(vecs, shape=(None, 3)).copy()
 
-        cdef const float[:, ::1] l_points = vecs
-        cdef unsigned int Np = l_points.shape[0]
-        self.thisptr.wrap(<vec3[float]*> &l_points[0, 0], Np)
+        if inplace is False:
+            cdef const float[:, ::1] l_points = vecs
+            cdef unsigned int Np = l_points.shape[0]
+            self.thisptr.wrap(<vec3[float]*> &l_points[0, 0], Np)
+        else:
+            cdef unsigned int Np = vecs.shape[0]
+            self.thisptr.wrap(<vec3[float]*> &vecs[0, 0], Np)
 
         return np.squeeze(vecs) if flatten else vecs
```

Comment:
If the calculation is done `inplace`, then you don't need to return anything; the user's expectation then is to use the array provided as an argument.